designs/2018-simplified-package-loading/README.md

```diff
@@ -0,0 +1,87 @@
+* This change is backwards-incompatible for end users who use global plugin installations or home-directory configs, which could cause some migration pain. (However, the change is compatible with existing shareable configs and plugins, so end users would be able to migrate without waiting for a third party to migrate.)
+* This change does not allow shareable configs to manage their own plugins, as proposed in [RFC #5](https://github.com/eslint/rfcs/pull/5). However, it is intended to be compatible with future solutions to that problem.
+
+## Backwards Compatibility Analysis
+
+* This change has no compatibility impact on shareable configs and plugins. Shareable configs already specify plugins as `peerDependencies`, implying that the plugins must be installed and loadable by the end user. This change simply updates ESLint to use that guarantee.
+* This is a breaking change for end users who use global ESLint installations and global plugin installations; the plugins must now be installed locally.
+* This is a breaking change for end users who use home-directory configs along with shareable configs or custom parsers; the shareable configs/parsers must be resolvable from the location of the home-directory config.
+* This is a breaking change for integration authors who depend on `Linter` loading custom parsers from the filesystem. These integration authors could explicitly use `Linter#defineParser` instead. (Most integration authors who need access to the filesystem are using `CLIEngine` anyway.)
+
+## Alternatives
+
+* Along with configs/parsers, ESLint could load plugins relative to the configs that depend on them. That would introduce the possibility of having two plugins with the same name, requiring a disambiguation mechanism for configuring that plugin's resources; see [RFC #5](https://github.com/eslint/rfcs/pull/5) for an example. This RFC is intended to be a simpler step to solve some problems, while allowing additional work to proceed on that problem if desirable.
+* To mitigate some compatibility impact, ESLint could fall back to the existing behavior when it fails to load a package using the new behavior, instead of throwing a fatal error. However, this would likely cause more confusion because users wouldn't understand why ESLint was finding their packages in certain cases, which would make it more difficult to debug problems. This fallback is only provided for the `espree` package (which falls back to the default parser if no user-installed version of the pacakge is found).
+
+## Open Questions
+
+* This proposal uses the CWD as the "package root" from which plugins are loaded. Are there other folders that would serve this purpose better (e.g. in cases where the `--config` flag is used to load a config from a separate project)?
```

Comment:
I'd think that specifying `--config` would require us to use the directory of that config file instead of any other.

Reply:
This might depend on the outcome of the other discussion in https://github.com/eslint/rfcs/pull/7#discussion_r244771759. It's worth noting that `--config` does not imply `--no-eslintrc`, i.e. there could be a a config specified with `--config` and also a different `root: true` config both used at the same time.

In the cases where a config file is within the user's project, I think it doesn't matter much whether the project root is the location of the config, because plugins should be resolved correctly anyway (although we should still decide on something for consistency). The case I was wondering about here is when `--config` crosses a `node_modules` boundary (e.g. `--config=node_modules/my-companys-configs/eslint/config.js`), in which case it's not guaranteed that resolving a plugin from the config will find the same result as resolving a plugin from the project.

In that case, it could be more useful to load plugins from the directory of that config file, because if someone is extracting all of their config files into a separate project, I imagine they would also want to extract their config's dependencies to that project to avoid having the same eslint-related dependencies in every project. On the other hand, this could create confusion when `--config` is used at the same time as a project-local config, because adding `--config` could break the project-local config. Ultimately, I don't think it's possible to always satisfy the expectations of both the project-local config author and the `--config` config author without allowing for different versions of the same plugin to be loaded simultaneously, which is intentionally not a part of this proposal.

I consider this to be an edge case where we could go either way. I'm leaning again towards always using the CWD, because this would avoid confusing scenarios where a local config breaks when a `--config` flag is added, and it's fairly easy to change the CWD if necessary.